src/datamodel_code_generator/input_model.py

```
@@ -0,0 +1,903 @@
+        return True
+    return source_family == output_family
+
+
+def _filter_defs_by_strategy(
+    schema: dict[str, Any],
+    nested_models: dict[str, type],
+    output_model_type: DataModelType,
+    strategy: InputModelRefStrategy,
+) -> dict[str, Any]:
+    """Filter $defs based on ref strategy, marking reused types with x-python-import."""
+    from datamodel_code_generator.arguments import InputModelRefStrategy as IRS  # noqa: PLC0415
```

Ruff (0.14.10)

Unused `noqa` directive (non-enabled: `PLC0415`)

Remove unused `noqa` directive

(RUF100)